packages/ember-simple-auth/tests/simple-auth/setup-test.js

```diff
@@ -168,6 +168,22 @@ describe('setup', function() {
         expect(this.authorizer.authorize).to.have.been.calledThrice;
       });
 
+      it('authorizes requests going to a foreign origin if the hosts on whitelist allow subdomain to origin', function() {
+        Configuration.crossOriginWhitelist = ['http://*.other-domain.com'];
+        setup(this.container, this.application);
+        Ember.$.get('http://test.other-domain.com/path/query=string');
+
+        expect(this.authorizer.authorize).to.have.been.calledOnce;
+
+        Ember.$.get('http://another-test.other-domain.com/path/query=string');
+
+        expect(this.authorizer.authorize).to.have.been.calledTwice;
+
+        Ember.$.get('http://test2.other-domain.com/path/query=string');
+
+        expect(this.authorizer.authorize).to.have.been.calledThrice;
+      });
```

Comment:
I added this tests, and works fine on browser. but on phantomjs I have problems with `*` in url, the hostname is empty

To reproduce: 
```` bash
$ echo "var link = document.createElement('a'); link.href='http://*.other-domain.com/mytest'; console.log(link.href) ; console.log(link.hostname)" | phantomjs

phantomjs> http://*.other-domain.com/mytest // <---href
// <--- hostname
0 // <--- port
```

Do you have any ideia for it?